packages/react-aria-components/src/Table.tsx

```diff
@@ -58,7 +58,7 @@ class TableCollection<T> extends BaseCollection<T> implements ITableCollection<T
       switch (node.type) {
         case 'column':
           columnKeyMap.set(node.key, node);
-          if (!node.hasChildNodes) {
+          if (node.firstChildKey == null) {
```

Comment:
this is technically what the old check was, is this still actually what we want to check, it's a bit of a strange check